terraform/aws/modules/gitlab/gitlab-route53.tf

```diff
@@ -8,14 +8,44 @@ data "kubernetes_service" "gitlab_nginx" {
   ]
 }
 
+
+data "aws_elb" "gitlab_nginx_elb" {
+  # hostname have the name of the LB: internal-a6c7cc5f1622422ea8cb712fe8s21b48-995323897.us-east-1.elb.amazonaws.com need to get the name in order to ge tthe zone_id that will be used to create the A record with alias
+  name = split("-", data.kubernetes_service.gitlab_nginx.load_balancer_ingress.0.hostname).1
+  depends_on = [
+    helm_release.gitlab,
+  ]
+}
+
 resource "aws_route53_record" "gitlab_record" {
   zone_id = var.private_hosted_zoneid
   name    = "gitlab"
-  type    = "CNAME"
-  ttl     = "60"
-  records = ["${data.kubernetes_service.gitlab_nginx.load_balancer_ingress.0.hostname}"]
+  type    = "A"
+
+  alias {
+    name                   = data.kubernetes_service.gitlab_nginx.load_balancer_ingress.0.hostname
+    zone_id                = data.aws_elb.gitlab_nginx_elb.zone_id
+    evaluate_target_health = false
+  }
+
   depends_on = [
     helm_release.gitlab,
   ]
 }
 
+
+resource "aws_route53_record" "registry_record" {
+  zone_id = var.private_hosted_zoneid
+  name    = "registry"
+  type    = "A"
+
+  alias {
+    name                   = data.kubernetes_service.gitlab_nginx.load_balancer_ingress.0.hostname
+    zone_id                = data.aws_elb.gitlab_nginx_elb.zone_id
```

Comment:
and this one to be:
`zone_id                = data.aws_elb_hosted_zone_id.main.id`

Reply:
works without this